### PR TITLE
Add rpc-proxy service for whitelisting JSON RPC methods to the sequencer.

### PR DESCRIPTION
**Pull request by optimisticben**
**Description**
Added an Nginx based proxy to restrict the json-rpc methods allowed to an upstream Sequencer.

**Additional context**
Uses Lua to parse json payload for whitelist comparison and producing Prometheus metrics.
Git wprkflow added to build rpc-proxy image with the same tag as the sequencer.

**Metadata**
Mostly based on https://github.com/adetante/ethereum-nginx-proxy

### CODE REVIEW

**Comment by tynes** [comment]
@optimisticben What is the status on this? Does it still need updates or is it ready for review?

File: ops/docker/rpc-proxy/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
```
Comment (tynes) [inline ops/docker/rpc-proxy/docker-entrypoint.sh:2]: This file could be cleaned up

File: ops/docker/rpc-proxy/docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+set -eo pipefail
+
+if [ -z "$SEQUENCER" ];then
+  echo "SEQUENCER env must be set, exiting"
+  exit 1
+fi
+
```
Comment (tynes) [inline ops/docker/rpc-proxy/docker-entrypoint.sh:9]: We probably want to make sure that `ETH_CALLS_ALLOWED` is set

**Review by tynes — CHANGES_REQUESTED** [review]
Once the docker entrypoint is cleaned up, this should be good to go

File: ops/docker/rpc-proxy/docker-entrypoint.sh (hunk 2 above)
Comment (optimisticben) [reply ops/docker/rpc-proxy/docker-entrypoint.sh:9]: added!

File: ops/docker/rpc-proxy/docker-entrypoint.sh (hunk 1 above)
Comment (optimisticben) [reply ops/docker/rpc-proxy/docker-entrypoint.sh:2]: Removed commented code.

**Review by gakonst — APPROVED** [review]

**Pull request merged**